### ISSUE TITLE
Add simple identity store to get started

### ISSUE THREAD

**Issue by arjantijms** [jakartaee/security#289]
The currently provided identity stores for LDAP and JDBC require some effort to populate.

Especially for people new to the platform, and even more so for example code, something simpler is required.

The biggest challenge is not in defining something simpler, but in getting it through the process. In the past there's always that one person who thinks it's insecure to have a simple store, and subsequently the proposal is withdrawn again.

**Comment by m-reza-rahman** [comment]
I've always tried to support this. I suggest committers vote on a formal proposal using the normal consensus process.

**Comment by hantsy** [comment]
I remember the original version included an `In Memory` implementation.

**Comment by arjantijms** [comment]
> I remember the original version had an `In Memory` implementation, but not included in the final version.

Indeed. It's still in Soteria, but someone raised a concern that it may not be secure, so it was left out of the spec.

See; https://github.com/eclipse-ee4j/soteria/blob/master/impl/src/main/java/org/glassfish/soteria/identitystores/EmbeddedIdentityStore.java

**Comment by hantsy** [comment]
It is good for demonstration and writing tests purpose. 

Spring Security also provides in-memory `Map` based userservice.
